timeflux/nodes/expression.py

```diff
@@ -108,27 +108,23 @@ def __init__(self, expr, eval_on, **kwargs):
         self._expr_ports = None
 
     def update(self):
-
         self.o.meta = self.i.meta
-
         if self._eval_on == 'ports':
             if self._expr_ports is None:
                 self._expr_ports = [port_name for port_name, _, _
-                                    in self.iterate()
+                                    in self.iterate('i*')
```

Comment:
Are you intending to match all inputs (including `i`) or only named inputs? If the latest, the convention must be enforced and the pattern must be `i_*`.

Reply:
ok done